etcdserver/raft.go

```diff
@@ -163,10 +163,19 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 
 					atomic.StoreUint64(&r.lead, rd.SoftState.Lead)
 					islead = rd.RaftState == raft.StateLeader
-					isCandidate = rd.RaftState == raft.StateCandidate
+					waitApply = rd.RaftState == raft.StateCandidate
 					rh.updateLeadership()
 				}
 
+				if !islead && !waitApply {
```

Comment:
can this go right above the `if waitApply { rh.waitForApply() }` code so the etcdserver apply doesn't have to wait on the entry scan?